contracts/NodeDelegator.sol

```diff
@@ -145,21 +146,28 @@ contract NodeDelegator is INodeDelegator, LRTConfigRoleChecker, PausableUpgradea
     {
         address lrtDepositPool = lrtConfig.getContract(LRTConstants.LRT_DEPOSIT_POOL);
 
-        bool success;
-        if (asset == LRTConstants.ETH_TOKEN) {
-            (success,) = payable(lrtDepositPool).call{ value: amount }("");
-        } else {
-            success = IERC20(asset).transfer(lrtDepositPool, amount);
+        // Convert any ETH to WETH before transferring
+        // The WETH address is different between chains so reading from the config.
+        // Ideally, the WETH address would be an immutable but following the existing pattern of using config for now.
+        address WETH_TOKEN_ADDRESS = lrtConfig.getLSTToken(LRTConstants.WETH_TOKEN);
+        if (asset == WETH_TOKEN_ADDRESS) {
+            uint256 ethBalance = address(this).balance;
+            if (ethBalance > 0) {
+                // Convert any ETH into WETH
+                IWETH(WETH_TOKEN_ADDRESS).deposit{ value: ethBalance }();
+            }
         }
 
+        bool success = IERC20(asset).transfer(lrtDepositPool, amount);
+
         if (!success) {
             revert TokenTransferFailed();
         }
     }
 
-    /// @notice Fetches balance of all assets staked in eigen layer through this contract
-    /// @return assets the assets that the node delegator has deposited into strategies
-    /// @return assetBalances the balances of the assets that the node delegator has deposited into strategies
+    /// @notice Fetches balance of all assets staked in EigenLayer through this contract. This includes LSTs and WETH.
+    /// @return assets the assets that the node delegator has deposited into EigenLayer.
+    /// @return assetBalances the balances of the assets that the node delegator has deposited into EigenLayer
     function getAssetBalances()
```

Comment:
Would it be possible to make `getAssetBalances` call `getAssetBalance` in a loop, which would un duplicate some code here?

Reply:
I initially thought the same but its more efficient this way as `getAssetBalances` calls `getDeposits` on the Eigen Strategy Manager which only returns assets if there is a balance. So for the Node Delegator that is managing ETH and not LSTs, it returns zero LSTs. The alternative is `getAssetBalances` calls `getSupportedAssetList` on `LRTConfig` and iterates over all 7 LSTs.

Here's the comparison with in the fork tests where one node has all the LSTs and the other just has native ETH

Calling `getDeposits` on Eigen Strategy Manager
```
| Function Name                                      | min             | avg    | median | max    | # calls |
| getAssetBalances                                   | 20491           | 81005  | 98013  | 124513 | 3       |
```

With `getSupportedAssetList` on LRTConfig
```
| Function Name                                      | min             | avg    | median | max    | # calls |
| getAssetBalances                                   | 99929           | 111382 | 107788 | 126429 | 3       |
```

Reply:
🔴 The current behavior of getAssetBalances will not report any ERC20's deposited to the Node until some of those tokens are deposited into eigenlayer.

The only place we use `getAssetBalances` is when removing a node. This would seem to be a time when gas doesn't matter. Even though it's not efficient, we could just loop it.

If we cared more about gas, we could do an internal method that both external ones use. It could take a weth address argument so we don't do multiple lookups. But I think we should just skip that for now and have cleaner code since that's far off the gas critical path.

Reply:
Good pickup. I'll change `getAssetBalances` to loop over `getAssetBalance`

Reply:
This change has been pushed